header/p2p/exchange_test.go

```diff
@@ -311,9 +311,11 @@ func createP2PExAndServer(t *testing.T, host, tpeer libhost.Host) (*Exchange, *h
 	require.NoError(t, err)
 	ex, err := NewExchange(host, []peer.ID{tpeer.ID()}, "private", connGater)
 	require.NoError(t, err)
-	ex.peerTracker.trackedPeers[tpeer.ID()] = &peerStat{peerID: tpeer.ID(), peerScore: 100}
 	require.NoError(t, ex.Start(context.Background()))
-
+	time.Sleep(time.Millisecond * 100) // give peerTracker time to add a trusted peer
+	ex.peerTracker.peerLk.Lock()
+	ex.peerTracker.trackedPeers[tpeer.ID()] = &peerStat{peerID: tpeer.ID(), peerScore: 100.0}
```

Comment:
Why do we need to wait if we add it manually?

Reply:
tpeer.ID is a trusted peer that is added on peerTracker starts. Here I re-insert it to the map.